web/docs/deployment/database.md

```diff
@@ -2,10 +2,66 @@
+In this section, we'll discuss what happens with the database when your app goes live. When you develop your app locally, you probably use the dev database that you start with the `wasp db start` command. However, when it's time to deploy your app, you'll need to set up a production database.
 
-TODO: the only requirement from Wasp's point of view - it has to be a PostgreSQL database accessible from the server via the `DATABASE_URL` environment variable
+### Production database requirements
 
-TODO: we'll talk about migrations: how are they created in development, how are they applied in production, how to debug failed migrations
+The server app that Wasp generates uses a PostgreSQL database. The only requirement from Wasp's point of view is that the database is accessible from the server via the `DATABASE_URL` environment variable.
 
-TODO: we'll talk about how to open the DB studio in development and production
\ No newline at end of file
+It can be a PostgreSQL database running on the same server as the server app, or it can be a managed PostgreSQL database service like [Fly Postgres](https://fly.io/docs/postgres/), [AWS RDS](https://aws.amazon.com/rds/), or some other service.
+
+## Migrations
+
+Every time you make a change in your [Prisma schema](../data-model/prisma-file.md) e.g. adding a new model, changing a field type, etc., you need to create a migration. Migrations are some code that describes the change you made in the schema, and they are used to apply the change to the database.
+
+The benefit of migrations is that you can apply the same change to multiple databases, e.g. your local development database and your production database. 
+
+### Creating migrations
+
+After you made a change in the Prisma schema, you can create a migration by running the following command:
+
+```bash
+wasp db migrate-dev
+```
+
+This command will create a new migration in the `migrations` directory. The migration is a set of SQL commands that describe the change you made in the schema.
+
+### Applying migrations
+
+**In development**, the migrations are applied as soon as you run the `wasp start` command. 
+
+**In production**, the server app first checks if there are any new migrations that need to be applied, and if there are, it applies them before starting the server. This way, the database schema is always in sync with the Prisma schema.
+
+The migrations might fail to apply if there is a conflict with the existing data in the database. In that case, you'll need to fix the migration and try again.
+
+### Debugging failed migrations
+
+If a migration fails to apply, the server app will log the error message and stop. You should then connect to the production database and see what went wrong. If you check the `_prisma_migrations` table, you'll see the failed migration there. 
+
+You can try resolving the erorr e.g. if you tried adding a `@unique` constraint to a field that already has duplicate values:
```

Comment:
Can we link to Prisma docs that have more details on this process? Handling failed migrations? And migrations in general? I don't mind that we explain it here a bit ourselves, but if there are materials about it in Prisma docs, I would like to them for more info.

Reply:
Looking at the Prisma docs, they rely heavily on `prisma` commands, I'm afraid if I link to those, people will start running the `prisma` commands and get stuck. https://www.prisma.io/docs/orm/prisma-migrate/workflows/patching-and-hotfixing#failed-migration

Reply:
Hm crap. I would still link to them I think, maybe warn them. But ok, whatever you think is best here is ok with me!